Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/remesh_planar_patches.h

```diff
@@ -0,0 +1,1494 @@
+ *  \param np_in an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+ *
+ *  \cgalNamedParamsBegin
+ *    \cgalParamNBegin{vertex_point_map}
+ *      \cgalParamDescription{a property map associating points to the vertices of `tm_in`}
+ *      \cgalParamType{a class model of `ReadablePropertyMap` with `boost::graph_traits<TriangleMeshIn>::%vertex_descriptor`
+ *                     as key type and `%Point_3` as value type}
+ *      \cgalParamDefault{`boost::get(CGAL::vertex_point, tm_in)`}
+ *      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ *                      must be available in `TriangleMeshIn`.}
+ *    \cgalParamNEnd
+ *  \cgalNamedParamsEnd
+ *
+ *  \param np_out an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below:
+ *
+ *  \cgalNamedParamsBegin
+ *    \cgalParamNBegin{vertex_point_map}
+ *      \cgalParamDescription{a property map associating points to the vertices of `tm_out`}
+ *      \cgalParamType{a class model of `WritablePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%vertex_descriptor`
+ *                     as key type and `%Point_3` as value type}
+ *      \cgalParamDefault{`boost::get(CGAL::vertex_point, tm_out)`}
+ *      \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+ *                      must be available in `TriangleMeshOut`.}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{face_patch_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each face the id
+ *                            of its patch in the range `[0, number of patches - 1]`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%face_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *    \cgalParamNBegin{vertex_corner_map}
+ *      \cgalParamDescription{a property map filled by this function and that will contain for each vertex its corner
+ *                            an id in the range `[0, number of corners - 1]`}
+ *      \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<TriangleMeshOut>::%vertex_descriptor`
+ *                     as key type and `std::size_t` as value type}
+ *      \cgalParamDefault{None}
+ *    \cgalParamNEnd
+ *  \cgalNamedParamsEnd
+ */
+template <typename TriangleMeshIn,
+          typename TriangleMeshOut,
+          typename FacePatchMap,
+          typename EdgeIsConstrainedMap,
+          typename VertexCornerMap,
+          typename NamedParametersIn = parameters::Default_named_parameters,
+          typename NamedParametersOut = parameters::Default_named_parameters>
+bool remesh_almost_planar_patches(const TriangleMeshIn& tm_in,
```

Comment:
` isotropic_remeshing()`   is in-place. Why does this function have an input and output mesh?

Reply:
This is not an incremental operation, the mesh is not updated but a new one is created.

Reply:
And do I get it right that the in and the out mesh can be the same?

Reply:
Yes thanks to the `visitor` named parameter